docs/src/views/CustomComponents.tsx

```diff
@@ -0,0 +1,235 @@
+import * as React from 'react'
+import * as cx from 'classnames'
+import * as _ from 'lodash'
+import { NavLink } from 'react-router-dom'
+import { Header } from 'semantic-ui-react'
+import {
+  Button,
+  Divider,
+  Provider,
+  createComponent,
+  ComponentSlotStyle,
+  ComponentVariablesInput,
+} from '@stardust-ui/react'
+
+import DocPage from '../components/DocPage/DocPage'
+import ExampleSnippet from '../components/ExampleSnippet/ExampleSnippet'
+
+interface StyledButtonProps {
+  className?: string
+  styles?: ComponentSlotStyle
+  variables?: ComponentVariablesInput
+  children?: React.ReactNodeArray | React.ReactNode
+}
+
+const StyledButton: React.SFC<StyledButtonProps> = createComponent<StyledButtonProps, {}>({
+  displayName: 'StyledButton',
+  render({ stardust, ...props }) {
+    const { classes } = stardust
+    const componentProps = _.pick(props, 'children')
+    return <button className={cx(props.className, classes.root)} {...componentProps} />
+  },
+})
+
+export default () => (
+  <DocPage title="Custom Components">
+    <Header as="h2" content="Overview" />
+    <p>
+      You can use your own components as part of the Stardust's theming mechanism. In order for all
+      theming aspects to be available to your custom components, you should use the{' '}
+      <code>createComponent</code>
+      function, provided by the Stardust library.
+    </p>
+
+    <Header as="h2" content="Create custom component" />
+    <p>
+      Let's take a look into one simple example of using the <code>createComponent</code> function
+      for adapting your custom component to the Stardust's theming mechanism.
+    </p>
+
+    <ExampleSnippet
+      value={[
+        `import { createComponent } from '@stardust-ui/react'`,
+        ``,
+        `export interface StyledButtonProps {`,
+        `  className?: string`,
+        `  styles?: ComponentSlotStyle`,
+        `  variables?: ComponentVariablesInput`,
+        `  children?: React.ReactNodeArray | React.ReactNode`,
+        `}`,
+        ``,
+        `const StyledButton: React.SFC<StyledButtonProps> = createComponent<StyledButtonProps, {}>({`,
+        `  displayName: 'StyledButton',`,
+        `  render: ({stardust, ...props}) => {`,
+        `    const { classes } = stardust`,
+        `    const componentProps = _.pick(props, 'children')`,
+        `    return (<button className={cx(props.className, classes.root)} {...componentProps} />)`,
+        `  }`,
+        `})`,
+      ].join('\n')}
+    />
+
+    <p>Let's go step by step throughout all bits from the renderComponent method.</p>
```

Comment:
~~`renderComponent`~~ ->  `createComponent`
I also changed this in the CHANGELOG.md file that seems to mention `renderComponent` as well
https://github.com/stardust-ui/react/pull/518/files#r235773446